app/services/course/material/download_service.rb

```diff
@@ -0,0 +1,64 @@
+class Course::Material::DownloadService
+  require 'zip'
+  require 'uri'
+  require 'open-uri'
+
+  class << self
+    # Download the contents and zip them.
+    #
+    # @param [Array<Course::Material|Course::Material::Folder>] contents The contents to be
+    #   downloaded.
+    # @return [String] The url to the zip file.
+    def download_and_zip(contents)
```

Comment:
Wait, I think there are some misunderstandings. 

So suppose I'm viewing folder A now.
I selected folder B and material M1, M2( all under folder A) and then click the download button of folder A. Then what should be the arguments ?